lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -158,10 +158,18 @@ def get(self):
                     },
                 }
                 url = self.param_template.sub("", url)
-                path = url[len(self.uri_prefix) + 1 :]
-                path = path.replace("/", "_")
-                apis[path] = urljoin(host, url)
-                templates[path] = template
+                path = rule.endpoint
+                # We have some URI endpoints that repeat a basic pattern, but
+                # only one should provide URI parameters. There may be a second
+                # without the parameter, and a third dropping the trailing '/'
+                # to provide full flexibility. We can only record one of these
+                # paths, and we'll always choose the one with parameters even
+                # if we don't see it first.
+                if path not in templates or (
+                    template["params"] and not templates[path]["params"]
+                ):
+                    apis[path] = urljoin(host, url)
+                    templates[path] = template
```

Comment:
You prefer the conditional/searching/dynamic approach over just requiring that the "right" one be listed first (or having a "primary" and optional "aliases", or some other way to distinguish it)?

Reply:
I'm going to move this comment to https://github.com/distributed-system-analysis/pbench/pull/2956.